torch/csrc/shim_common.cpp

```diff
@@ -109,8 +110,12 @@ static c10::IValue to_ivalue(
           stable_ivalue, extension_build_version));
     }
     case c10::TypeKind::DeviceObjType: {
-      return c10::IValue(torch::stable::detail::_to<c10::Device>(
-          stable_ivalue, extension_build_version));
+      auto deleter = [](DeviceOpaque* device) { torch_delete_device(device); };
```

Comment:
I need to confirm the idea I have works (I'm going to do it for StableList), but in essence it would be to make a new reference from the devicehandle, and then explicitly delete the old devicehandle. No need to change code here until I confirm it works.